go/vt/vtgate/tabletgateway.go

```diff
@@ -282,7 +295,9 @@ func (gw *TabletGateway) withRetry(ctx context.Context, target *querypb.Target,
 		if len(tablets) == 0 {
 			// if we have a keyspace event watcher, check if the reason why our primary is not available is that it's currently being resharded
 			// or if a reparent operation is in progress.
-			if kev := gw.kev; kev != nil {
+			// We only check for whether reshard is ongoing or primary is serving or not, only if the target is primary. We don't want to buffer
+			// replica queries, so it doesn't make any sense to check for resharding or reparenting in that case.
+			if kev := gw.kev; kev != nil && target.TabletType == topodatapb.TabletType_PRIMARY {
```

Comment:
Oh i had in mind to mark this specifically as unrelated, but forgot about it. We don't buffer replica queries even now. All the subsequent function calls in the if checks separately also check for the target being Primary or not. For replicas we were not buffering before, and we aren't now either. In some sense this change is ineffectual, but makes it easier to read the code instead of checking each individual function to see how they handle replicas.